pkg/router/metrics/haproxy/haproxy.go

```diff
@@ -428,6 +444,10 @@ loop:
 			return
 		}
 
+		if rows == 0 && len(e.counterIndices) < len(row) {
+			e.counterIndices = append(e.counterIndices, make([]byte, len(row)-len(e.counterIndices))...)
```

Comment:
I could buy the argument that if the field count ever differed we should panic (that was not the case at the time the code was written, because rhel could pick up an update and we didn't control that process in our CI).

I'd be in favor of a follow up that panicked if the expected metrics do not match, but we would have to be certain that we would catch that in our CI before releasing the router image.

Reply:
> I could buy the argument that if the field count ever differed we should panic (that was not the case at the time the code was written, because rhel could pick up an update and we didn't control that process in our CI).

It seemed weird to me that we update `e.counterIndices` but not `e.counterIndexSize`, and then I realized that `exportCSVFields` does not perform a bounds check when it indexes `e.counterIndices` but rather assumes that the metric index is in bounds, and that led me to wonder why we even needed to update `e.counterIndices`.

A panic would cause a crashloop because the router will just come back up expecting the same number of columns and most likely getting the unexpected number from HAProxy, right?

> I'd be in favor of a follow up that panicked if the expected metrics do not match, but we would have to be certain that we would catch that in our CI before releasing the router image.

Would a unit test suffice?